### ISSUE TITLE
Umbraco 13 (version 13.0.3), Grid (legacy) editor - Rich Text editor controls not rendering/available in Classic mode

### ISSUE THREAD

**Issue by PeatLaur** [umbraco/Umbraco-CMS#15515]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.0.3

### Bug summary

Using the Legacy Grid and macros (one of the reasons to update is to move to Block Grid). The first section of RTE content in the back office renders the RTE controls fine, but the controls are not rendered in the subsequent RTE sections of the page.

Example - first RTE renders the controls fine
![umb13-rte-missing-in-grid](https://github.com/umbraco/Umbraco-CMS/assets/48582574/9bdd6394-2934-43e9-8e81-544fca0c227f)

NO controls on subsequent RTE editors down the page:
![umb13-rte-missing-in-grid-2](https://github.com/umbraco/Umbraco-CMS/assets/48582574/aed55ef8-9ac9-43ef-b60f-20cd0b4d8668)

Example 2:
![umb13-rte-top-rte-avaliable-grid](https://github.com/umbraco/Umbraco-CMS/assets/48582574/e0ba3788-4716-4589-98b2-d6517017d954)

If I change to inline, then the RTE controls to display when some text is highlighted:
![umb13-rte-inline-in-grid](https://github.com/umbraco/Umbraco-CMS/assets/48582574/55048b8d-e945-4a54-83ce-0c1adb189f80)

We are upgrading from Umbraco 10.8.0 to Umbraco 13.0.3 as a separate project, with the aim to move over to Umbraco in the coming months. I take a copy of our Umbraco 10 database and upgrade it to Umbraco 13 with at first it seems are no issues.

Then, in trying to edit one of the existing CMS pages, built using the Legacy Grid and macros (one of the reasons to update is to move to Block Grid).  WE wish to be able to still use the legacy Grid editor and the RTE editor within it while we move pages over to Block Gris and CSS Grid.

### Specifics

Happens on all back office pages where we use Grid (legacy) to edit content.
Happens on all browsers.
If I change to inline, then the RTE controls to display when some text is highlighted.
Our content editors are used to working with the Classic style of RTE editing, and we wish to continue to be able to use it.

### Steps to reproduce

Install Umbraco 13.0.3, upgrade an existing Umbraco 10 database which uses Legacy Grid and RTE editors within the grids.
Start to edit the pages.
On some occasions, if I added a new rich text area and saved the page, when it reloaded all the RTE controls were rendered. but I can go to another page, then go back to the page I have just rendered and once again, none or only the top most RTE block controls on the page were rendered and ready to use.

### Expected result / actual result

Expected the RTE editor controls to render in Classic mode on all RTE parts of a page compiled using the Grid (legacy).

---
_This item has been added to our backlog AB#36592_

**Comment by PeatLaur** [comment]
![umb13-rte-tox-editor-fadeout-class](https://github.com/umbraco/Umbraco-CMS/assets/48582574/59d119b4-38a5-4edf-abb8-481091972125)

It looks like the tox-editor-fadeout class is not being removed on the RTE editor controls div.

**Comment by Zeegaan** [comment]
I can certainly reproduce this 🐛

**Comment by iOvergaard** [comment]
Hi, I just wanted to give a status here; we are working on this, but the problem proves difficult to locate. What we know so far is that it is most likely to happen if the grid is placed on the first content tab. If you for whatever reason place the grid on a 2nd tab, I haven't been able to reproduce it suggesting, that it's a race condition somewhere between the AngularJS initialization and TinyMCE. Another thing is that the editor renders but doesn't show its toolbar (just like @PeatLaur pointed out in their last comment) suggesting that the configuration of tinymce is not always "correct".

We'll continue looking into this problem.

**Comment by iOvergaard** [comment]
Some more troubleshooting have revealed that it is in fact caused by having a sticky toolbar, which for some reason do not work inside a grid layout editor.

As a workaround, setting this generally in appsettings to `false` should at least allow users to continue their work:

```json
"Umbraco": {
  "CMS": {
    "RichTextEditor": {
      "CustomConfig": { "toolbar_sticky": false }
    }
  }
}
```

**Comment by iOvergaard** [comment]
This should be fixed with #15595 which we expect to release with Umbraco 13.2. Thanks for reporting this! #h5yr

**Comment by PeatLaur** [comment]
Hi Zeegaan and iOvergaard and any one who contributed in resolving this issue. Thank you. I look forward to Umbraco 13.2 and being able to upgrade our website from Umbraco 10 before starting the migration of our content from Grid to Block Grid.
I will of course also look at the config change Rich Text Editor custom config in the meantime. Kind regards.

**Comment by iOvergaard** [comment]
@PeatLaur We have "upgraded" this issue to 13.1, so it should be fixed already later this week when 13.1 is out. Stay tuned!

**Comment by RiaanP** [comment]
To add a weird little quirk to this bug, which I am experiencing in an Umbraco 11.5.0 site, is that when I click in the text content of the RTE, type some garbage characters and then hit Ctrl+Z on the keyboard, the controls all appear. Discovered it completely by accident, but at least it helped us get some content published while we could find a deployment window. Maybe it can help someone as well in the interim.

1. Click in the text of your RTE.
2. Type a few characters
3. Press Ctrl+Z on your keyboard (undo)
